### PR TITLE
Fix plots logging

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Improves Ultralytics Platform/HUB training uploads by deduplicating plots, shrinking curve payloads, polishing logs, and reducing noisy GPU-monitoring warnings 📉🚀

### 📊 Key Changes
- Added `_interp_plot()` to downsample large curve datasets (e.g., PR curves) from ~1000 points to 101 points before uploading, reducing storage and payload size 📦
- Deduplicated plots by `type` in two places:
  - `Validator.on_plot()` now skips registering duplicate plot types 🧹
  - Platform `on_train_end()` merges trainer + validator plots into a single “one per type” set ✅
- Updated Platform logging with a colored `"Platform: "` prefix and direct “view results” URL messaging (pointing to `alpha.ultralytics.com`) 🔗
- Improved NVIDIA NVML initialization behavior in `SystemLogger`:
  - Skips NVML init on macOS 🍎
  - Only warns about NVML init failures when CUDA is actually available, reducing unnecessary warnings 🔇
- Adjusted PR curve upload format: PR curve `y` is now transposed to `y[class][point]` to align with other curve formats and simplify frontend handling 🔄
- Documentation updated to include the new `ultralytics.utils.callbacks.platform._interp_plot` reference entry 📝

### 🎯 Purpose & Impact
- Faster and lighter result uploads to Ultralytics Platform/HUB due to smaller curve data and fewer duplicate plots ⚡
- Cleaner, more consistent interactive plotting data for the frontend (especially PR curves) 📈
- Reduced confusion and log spam for users without NVML support (or on macOS), while still warning appropriately on CUDA systems 🛠️
- Clearer UX with direct “Streaming to …” and “View results at …” links during/after training 🔎

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist (this is automated; an engineer will assist soon) 🧩✅

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Overall, the PR looks solid and addresses the duplicate/oversized plot logging problem in a clear, maintainable way. Two areas need attention: `_init_nvidia()` can return `None` on a common failure path (should always return `False` on failure), and the new plot dedup/collection logic can silently drop plots missing a `type`. Additionally, `_interp_plot()` should defensively handle 2D `y` orientation to avoid incorrect interpolation or runtime errors for non-PR curve plots.

💬 Posted 3 inline comments

File: ultralytics/utils/logger.py [hunk 1]
```diff
@@ -330,14 +330,19 @@ def __init__(self):
...
+
         try:
-            assert not MACOS
             check_requirements("nvidia-ml-py>=12.0.0")
             self.pynvml = __import__("pynvml")
             self.pynvml.nvmlInit()
             return True
         except Exception as e:
-            LOGGER.warning(f"SystemLogger NVML init failed: {e}")
+            import torch
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/logger.py:342]: ⚠️ **HIGH**: `_init_nvidia()` can now return `None` when NVML init fails but CUDA is not available, because the `except` block only `return False` inside the `if torch.cuda.is_available()` branch. Callers expecting a strict `bool` may mis-handle `None` (e.g., truthiness checks, serialization). Ensure the function returns `False` on all failure paths.

File: ultralytics/utils/callbacks/platform.py [hunk 2]
```diff
@@ -31,6 +32,34 @@
...
+    if len(x) <= n:
+        return plot  # Already small enough
+
+    # New x values (101 points gives clean 0.01 increments: 0, 0.01, 0.02, ..., 1.0)
+    x_new = np.linspace(x[0], x[-1], n)
+
+    # Interpolate y values (handle both 1D and 2D arrays)
+    if y.ndim == 1:
+        y_new = np.interp(x_new, x, y)
+    else:
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/callbacks/platform.py:52]: 💡 **MEDIUM**: `_interp_plot()` assumes that for 2D `y`, iterating `for yi in y` yields per-series arrays aligned with `x` (i.e., `y.shape == (n_series, n_points)`). If a plot provides `y` as `(n_points, n_series)` (common for curve outputs), `np.interp(x_new, x, yi)` will receive `yi` with length `n_series` and fail or produce incorrect interpolation. Consider detecting orientation via `y.shape` vs `len(x)` and interpolating along the axis that matches `x`.

File: ultralytics/utils/callbacks/platform.py [hunk 3]
```diff
@@ -272,11 +303,15 @@ def on_train_end(trainer):
...
 
-    # Collect plots from trainer and validator
-    plots = [info["data"] for info in getattr(trainer, "plots", {}).values() if info.get("data")]
-    plots += [
-        info["data"] for info in getattr(getattr(trainer, "validator", None), "plots", {}).values() if info.get("data")
-    ]
+    # Collect plots from trainer and validator, deduplicating by type
+    plots_by_type = {}
+    for info in getattr(trainer, "plots", {}).values():
+        if info.get("data") and info["data"].get("type"):
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/callbacks/platform.py:309]: 💡 **MEDIUM**: Plot collection now ignores any plot payloads that don't include `data['type']`. Previously, any `info['data']` was forwarded. If any existing or future plots omit `type` (or if `type` is accidentally missing), they'll be silently dropped from uploads. Consider either preserving untyped plots or adding a fallback key (e.g., by filename) so they still get uploaded.

**Comment by UltralyticsAssistant** [comment]
Huge congrats on the merge, @glenn-jocher — this is a seriously thoughtful set of refinements that will be felt by everyone training with Ultralytics Platform and Ultralytics HUB.

> “Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away.” — **Antoine de Saint‑Exupéry**

This PR embodies that mindset: lighter uploads via curve downsampling, cleaner results via plot deduplication, more consistent PR curve formatting for the frontend, and a noticeably calmer training experience thanks to smarter NVML warnings (especially for macOS users). The polished `"Platform: "` logging and direct “view results” link are the kind of UX details that quietly elevate the whole workflow.

Thanks for pushing the platform experience forward with changes that are both practical and elegantly minimal.
